lib/spack/spack/package.py

```diff
@@ -22,35 +23,34 @@
 import sys
 import textwrap
 import time
-from six import StringIO
-from six import string_types
-from six import with_metaclass
-from ordereddict_backport import OrderedDict
 
-import llnl.util.tty as tty
+import six
 
-import spack.config
-import spack.paths
-import spack.store
+import llnl.util.tty as tty
 import spack.compilers
-import spack.directives
+import spack.config
 import spack.dependency
+import spack.directives
 import spack.directory_layout
 import spack.error
 import spack.fetch_strategy as fs
 import spack.hooks
 import spack.mirror
 import spack.mixins
 import spack.multimethod
+import spack.paths
 import spack.repo
+import spack.store
 import spack.url
 import spack.util.environment
 import spack.util.web
-import spack.multimethod
-
 from llnl.util.filesystem import mkdirp, touch, working_dir
 from llnl.util.lang import memoized
 from llnl.util.link_tree import LinkTree
+from ordereddict_backport import OrderedDict
+from six import StringIO
+from six import string_types
+from six import with_metaclass
```

Comment:
We should talk about import order at some point. According to flake8, it should be:

1. future imports
2. builtin imports
3. third-party libraries
4. local files

Within each section, order isn't specified by PEP-8, but tools like `flake8-import-order` assert alphabetical order. We should consider adding that flake8 extension to our CI

Reply:
But obviously, it doesn't matter for this particular PR, just want to come up with an agreed upon standard at some point.

Reply:
I'd like to use `black` as soon as we drop Python 2.6 support. If we do that there will be nothing to agree on :laughing:

Reply:
How exactly will black work? Will it be an action/bot that adds an additional commit after each commit (if necessary)? Can you point me to a repo where you have black setup?

Reply:
Why do we need to wait until we drop 2.6 support to use black? Can't we have it running in a 3.8 action/bot?

Reply:
> Why do we need to wait until we drop 2.6 support to use black? Can't we have it running in a 3.8 action/bot?

@adamjstewart Apologies, I had already resolved this tab and didn't notice your question. We need to drop Python 2.6 first because black will reformat the code automatically and has language compatibility options that go back to Python 2.7 (so the re-formatted code is not granted to work on earlier version of Python).

> How exactly will black work? Will it be an action/bot that adds an additional commit after each commit (if necessary)? Can you point me to a repo where you have black setup?

Here we use black during CI https://github.com/archspec/archspec I set it up to error if any file needs modifications since I don't like automatic commits to occur because of CI.